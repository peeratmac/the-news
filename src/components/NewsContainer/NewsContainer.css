.news-container {
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-gap: 50px;
  margin-right: 30px;
}
