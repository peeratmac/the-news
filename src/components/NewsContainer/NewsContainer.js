import React from 'react';
import './NewsContainer.css';
import NewsArticle from '../NewsArticle/NewsArticle';
import PropTypes from 'prop-types';

const NewsContainer = ({ articles }) => {
  return articles.map(article => (
    <NewsArticle
      key={article.id}
      headline={article.headline}
      image={article.img}
      description={article.description}
      url={article.url}
    />
  ));
};

export default NewsContainer;

NewsContainer.propTypes = {
  articles: PropTypes.array.isRequired
};
