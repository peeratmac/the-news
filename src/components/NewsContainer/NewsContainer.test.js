import React from 'react';
import { shallow } from 'enzyme';
import NewsContainer from './NewsContainer';

describe('NewsContainer', () => {
  it('should match the NewsContainer Snapshot', () => {
    const mockSelectedNews = [
      {
        id: 1,
        headline:
          'Unfixable iOS Device Exploit Is the Latest Apple Security Upheaval',
        img:
          'https://media.wired.com/photos/5d8e600f34cab0000861c48d/master/w_2560%2Cc_limit/security_jailbreak_MADEPD.jpg',
        description:
          "Any iPhone device from 2011 to 2017 could soon be jailbroken, thanks to an underlying flaw that there's no way to patch.",
        url: 'https://www.wired.com/story/ios-exploit-jailbreak-iphone-ipad/'
      },
      {
        id: 2,
        headline:
          'SpaceX’s orbital Starship prototype comes together ahead of update event',
        img:
          'https://techcrunch.com/wp-content/uploads/2019/09/EFfmFEEU8AA1SmA.jpeg?w=1390&crop=1',
        description:
          'SpaceX has completed joining the bottom and top halves of the Starship orbital-scale prototype it’s building in Boca Chica, Texas.',
        url:
          'https://techcrunch.com/2019/09/27/spacexs-orbital-starship-prototype-comes-together-ahead-of-update-event/'
      }
    ];
    const wrapper = shallow(<NewsContainer articles={mockSelectedNews} />);
    expect(wrapper).toMatchSnapshot();
  });
});
