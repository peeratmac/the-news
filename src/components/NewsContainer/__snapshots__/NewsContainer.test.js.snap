// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsContainer should match the NewsContainer Snapshot 1`] = `
Array [
  <NewsArticle
    description="Any iPhone device from 2011 to 2017 could soon be jailbroken, thanks to an underlying flaw that there's no way to patch."
    headline="Unfixable iOS Device Exploit Is the Latest Apple Security Upheaval"
    image="https://media.wired.com/photos/5d8e600f34cab0000861c48d/master/w_2560%2Cc_limit/security_jailbreak_MADEPD.jpg"
    key="1"
    url="https://www.wired.com/story/ios-exploit-jailbreak-iphone-ipad/"
  />,
  <NewsArticle
    description="SpaceX has completed joining the bottom and top halves of the Starship orbital-scale prototype it’s building in Boca Chica, Texas."
    headline="SpaceX’s orbital Starship prototype comes together ahead of update event"
    image="https://techcrunch.com/wp-content/uploads/2019/09/EFfmFEEU8AA1SmA.jpeg?w=1390&crop=1"
    key="2"
    url="https://techcrunch.com/2019/09/27/spacexs-orbital-starship-prototype-comes-together-ahead-of-update-event/"
  />,
]
`;
